external-crates/move/crates/move-binary-format/src/normalized.rs

```diff
@@ -505,6 +513,33 @@ impl<S> Datatype<S> {
         }
     }
 
+    /// Case for generating a `Datatype` to write down on a Struct or Enum definition
+    pub fn new_for_data_definition<Pool: StringPool<String = S>>(
+        pool: &mut Pool,
+        m: &CompiledModule,
+        idx: DatatypeHandleIndex,
+    ) -> Self {
+        let datatype_handle = m.datatype_handle_at(idx);
+        let defining_module_handle = m.module_handle_at(datatype_handle.module);
+        let datatype_name = pool.intern(m.identifier_at(datatype_handle.name));
+        let defining_module_address = *m.address_identifier_at(defining_module_handle.address);
+        let defining_module_name = pool.intern(m.identifier_at(defining_module_handle.name));
+        let type_arguments = datatype_handle
+            .type_parameters
+            .iter()
+            .enumerate()
+            .map(|(ndx, _)| Type::TypeParameter(ndx as u16))
```

Comment:
I'm not a huge fan of this as it is a bit nonsensical? Like these type parameters are not bound anywhere, but maybe I'm missing something

Reply:
I am really not sure what is up with this PR -- I rebased it at some point and a bunch of weird code that I deleted reappeared. This code should not even be here.